src/docs/options.md

```diff
@@ -55,58 +55,87 @@ levels in different tasks.
 Option Types
 ------------
 
-Option value literal formats are the same on the command-line, environment variables and in `pants.ini`,
-with a couple of extra command-line conveniences.
+Option value formats are the same on the command-line, environment variables and in `pants.toml`, except for list options. There are also a couple of extra command-line conveniences.
 
 ### Boolean Options
 
-The boolean option values are: `false`, `False`, `true`, `True`.
+The boolean option values are `false` and `true`.
 
-On the command line, you can omit the value thus: `--foo-bar` is the same as `--foo-bar=false`,
+With the command line and environment variables, you can omit the value thus: `--foo-bar` is the same as `--foo-bar=false`,
   and `--no-foo-bar` is the same as `--foo-bar=false`.
 
 ### Int and Float Options
 
 Specify values in decimal notation: `--foo=5`, `--bar=4.5`.
 
+    :::toml
+    foo = 5
+    bar = 4.5
+
 ### String Options
 
-Surround strings with single or double quotes if they contain embedded spaces: `--foo="hello, world"`.
+Surround strings with single or double quotes: `--foo="hello, world"`.
+
+    :::toml
+    foo = "hello world"
 
 ### List Options
 
-List options can be appended to and filtered, as well as overridden.
-For example, for an option `--foo` whose default value is `[1, 2]`, then in `pants.ini`:
+List options can be added to and removed, as well as overridden.
+For example, for an option `--foo` whose default value is `[1, 2]`, then in `pants.toml`:
+
++ `foo = 3` will yield `[1, 2, 3]`.
++ `foo.add = [3, 4]` will yield `[1, 2, 3, 4]`.
++ `foo.remove = [1]` will yield `[2]`.
++ `foo = [3, 4]` will yield `[3, 4]`.
+
+In `pants.toml`, you may both add and remove simultaneously:
+
+    :::toml
+    # This will yield [2, 3, 4]
+    foo.add = [3, 4]
+    foo.remove = [1]
 
-+ `foo: 3` will yield `[1, 2, 3]`.
-+ `foo: +[3, 4]` will yield `[1, 2, 3, 4]`.
-+ `foo: -[1]` will yield `[2]`.
-+ `foo: [3, 4]` will yield `[3, 4]`.
+Adding a single value and replacing a list works the same as `pants.toml` for the command line and environment variables.
 
-Multiple append and filter expressions may be delimited with commas,
-allowing you to append and filter simultaneously:
++ `--foo=3` will yield `[1, 2, 3]`.
++ `--foo='[3, 4]'` will yield `[3, 4]`.
 
-+ `foo: +[3,4],-[1]` will yield `[2, 3, 4]`.
+The syntax is a little different from `pants.toml`, however, when removing values or adding multiple values.
```

Comment:
This potentially points out a wart that could be rectified later. It may make sense to be uniform and use `PANTS_CONFIG_FILES.ADD` (or `PANTS_CONFIG_FILES_ADD`) and `--pants-config-files.add` (`--pants-config-files-add`).

Reply:
I would love for us to have either of those alternatives. Not sure which I'd vote for yet.